bitnami/kiam/templates/_helpers.tpl

```diff
@@ -0,0 +1,91 @@
+{{/*
+Return the proper kiam image name
+*/}}
+{{- define "kiam.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "kiam.imagePullSecrets" -}}
+{{- include "common.images.pullSecrets" (dict "images" (list .Values.image) "global" .Values.global) -}}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use (server)
+*/}}
+{{- define "kiam.server.serviceAccountName" -}}
+{{- if .Values.server.serviceAccount.create -}}
+    {{ default (printf "%s-server" (include "common.names.fullname" .)) .Values.server.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.server.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use (agent)
+*/}}
+{{- define "kiam.agent.serviceAccountName" -}}
+{{- if .Values.agent.serviceAccount.create -}}
+    {{ default (printf "%s-agent" (include "common.names.fullname" .)) .Values.agent.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.agent.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Generate certificates for kiam agent and server
+*/}}
+{{- define "kiam.agent.gen-certs" -}}
+{{- $ca := .ca | default (genCA "kiam-ca" 365) -}}
+{{- $_ := set . "ca" $ca -}}
+{{- $cert := genSignedCert "Kiam Agent" nil nil 365 $ca -}}
+{{ .Values.agent.tlsCerts.caFileName }}: {{ $ca.Cert | b64enc }}
+{{ .Values.agent.tlsCerts.certFileName }}: {{ $cert.Cert | b64enc }}
+{{ .Values.agent.tlsCerts.keyFileName }}: {{ $cert.Key | b64enc }}
+{{- end -}}
+
+{{- define "kiam.server.gen-certs" -}}
+{{- $altNames := list (printf "%s-server" (include "common.names.fullname" .)) (printf "%s-server:443" (include "common.names.fullname" .)) "127.0.0.1:8443" -}}
```

Comment:
Why are we hardcoding the service ports here? Could we use the parameters specified in `values.yaml` for container port?

Reply:
Good catch

Reply:
Fixed